.github/workflows/openssl_versions.yml

```diff
@@ -0,0 +1,175 @@
+---
+name: OpenSSL Versions
+
+on:
+  pull_request:
+    branches: ["main"]
+  push:
+    branches: ["main"]
+
+jobs:
+  build:
+    name: Versions
+    runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false
+      matrix:
+        name: [ossl30, ossl32, ossl35, debian, centos10]
+        include:
+          - name: ossl30
+            build: true
+            version: "3.0"
+            repotype: "dnf5"
+            repocachepath: "/var/cache/libdnf5"
+            container: fedora:latest
+          - name: ossl32
+            build: true
+            version: "3.2"
+            repotype: "dnf5"
+            repocachepath: "/var/cache/libdnf5"
+            container: fedora:latest
+          - name: ossl35
+            build: true
+            version: "3.5"
+            repotype: "dnf5"
+            repocachepath: "/var/cache/libdnf5"
+            container: fedora:latest
+          - name: debian
+            build: false
+            version: "3.5"
+            repotype: "apt"
+            container: debian:testing
+          - name: centos10
+            build: false
+            version: "3.5"
+            repotype: "dnf5"
```

Comment:
the only nit is that the centos10 does not have dnf5 so I would prefer now changing the dnf5 in all the `repotype` to `dnf`  (or just removing it and using the `repocachepath`